src/main/java/com/cloudbees/hudson/plugins/folder/computed/FolderComputation.java

```diff
@@ -195,20 +208,44 @@ public File getEventsFile() {
         return new File(folder.getComputationDir(), "events.log");
     }
 
-    public TaskListener createEventsListener() throws IOException {
+    @WithBridgeMethods(TaskListener.class)
+    @Nonnull
+    public synchronized StreamTaskListener createEventsListener() {
         File eventsFile = getEventsFile();
-        FileUtils.forceMkdir(eventsFile.getParentFile());
-        boolean rotate = eventsFile.length() > EVENT_LOG_MAX_SIZE * 1024;
-        OutputStream os;
-        if (BACKUP_LOG_COUNT != null) {
-            os = new ReopenableRotatingFileOutputStream(eventsFile, BACKUP_LOG_COUNT);
-            if (rotate) {
-                ((ReopenableRotatingFileOutputStream) os).rewind();
-            }
-        } else {
-            os = new FileOutputStream(eventsFile, !rotate);
+        if (!eventsFile.getParentFile().isDirectory() && !eventsFile.getParentFile().mkdirs()) {
+            LOGGER.log(Level.WARNING, "Could not create directory {0} for {1}",
+                    new Object[]{eventsFile.getParentFile(), folder.getFullName()});
+            // TODO return a StreamTaskListener sending output to a log, for now this will just try and fail to write
+        }
+        if (eventStreams == null) {
+            eventStreams = new EventOutputStreams(new EventOutputStreams.OutputFile() {
+                @NonNull
+                @Override
+                public File get() {
+                    return getEventsFile();
+                }
+
+                @Override
+                public boolean canWriteNow() {
+                    GregorianCalendar timestamp = new GregorianCalendar();
+                    timestamp.setTimeInMillis(System.currentTimeMillis() - 10000L);
+                    Queue.Item probe = new Queue.WaitingItem(timestamp, folder, Collections.<Action>emptyList());
+                    for (QueueTaskDispatcher d: QueueTaskDispatcher.all()) {
+                        if (d.canRun(probe) != null) {
```

Comment:
Ok! 

1. `Queue.BlockedItem.getCauseOfBlockage()` calls this method (does not use any lock to guard)
2. `Queue.Item.getWhy()` calls `Queue.BlockedItem.getCauseOfBlockage()` (does not use any lock to guard) and is `@Exported`

Thus there are already code paths that execute the `QueueTaskDispatcher.canRun(Queue.Item)` without holding the `Queue` lock.

Thus I deem it no less safe than the current usage.

I have filed [JENKINS-42248](https://issues.jenkins-ci.org/browse/JENKINS-42248) to enhance core with a less race-condition prone way to handle locking the file-system